R/dependencies.R

```diff
@@ -1707,6 +1707,22 @@ renv_dependencies_report <- function(errors) {
 
 }
 
+
+renv_dependencies_param_is_true <- function(x, default = TRUE) {
+  # https://github.com/rstudio/renv/issues/1558
+  if (is.call(x) || is.symbol(x)) {
+    x <- tryCatch(renv_dependencies_eval(x), error = identity)
+    if (inherits(x, "error"))
+      return(default)
+  }
+
+  if (is.logical(x) && length(x) == 1 && !is.na(x)) {
```

Comment:
For a chunk header like:

`````
```{r, renv.ignore=0}
# blah
```
`````

We would end up ignoring the value of `0` as it's not logical?

In other words, I think we should fall back to `truthy()` here instead for whatever value the user has supplied, or just coerce non-logical values to logical.

Reply:
Sorry, I didn't realise knitr was so loosely typed here. Even `include = "false"` works 😭

Reply:
"Be liberal in what you accept", as they say. 😉

Reply:
FWIW `truthy()` isn't quite right either because `include = "True"` doesn't work. But we can coerce to logical.